amads/time/meter/break_it_up.py

```diff
@@ -365,9 +484,8 @@ def start_hierarchy_from_ts(ts_str: str, minimum_pulse: int = 64):
 
 
 def starts_from_ts_and_levels(
-    ts_str: str,  # TODO Union[str, meter.time_signature] or similar
+    ts_str: str,
```

Comment:
I think we had agreed to use full names rather than abbreviations in Python expressions. Do you think that is impractical here?

Reply:
It's at least borderline for me. Using some conventions like ts for time signature and dur for duration seems reasonable. Philosophically, there's a reason we use abbreviations, even  in formal writing, etc. :-)

Reply:
I think often there are ways around it, though. For example here I think the problem is that we are trying to code the input type into the function name. This would be a natural application for OOP, I think:

```py
class TimeSignature:
    def get_starts(levels=None):
        ...
```

I think now that IDEs have autocomplete there is little overhead for using full words, and it does generally enhance readability. `dur` is normally quite obvious from context but I wasn't sure what `ts` meant until I read the documentation. This matters particularly in a package when we are taking a liberal approach to namespace exporting and so the context is broad.

Reply:
Yes that's all fine:
- no rush with the review.
- continue discussion via email.
- probably full names at least for anything not immediately obvious.

One objection (possibly trivial): the notion of a `TimeSignature` class. As I see it, time signatures are either trivial (just two numbers?), or basic/under-defined metrical structures, or else not in scope here. Here as elsewhere, I find the maxim "we're not in the business of making notation" a helpful line in the sand.

Reply:
Yes, in this case (having now looked in the code) this is actually the only use of time signature, more or less, so creating a time signature class would be overkill. I think this also means, though, that abbreviating time signature is a bad idea because it would be an abbreviation that only gets used for one function